lib/puppet/provider/package/apt.rb

```diff
@@ -42,7 +42,7 @@ def self.instances
 
   def query
     hash = super
-    hash[:mark] = :manual if aptmark('showmanual').split("\n").include?(@resource[:name])
+    hash[:mark] = :manual if @resource.present?(@resource) && aptmark('showmanual', @resource[:name]).strip == @resource[:name]
```

Comment:
we could also use the current ensure value from hash as trigger for `aptmark`, since may better represent the package state on the OS level

Reply:
Something like this I guess it should work
```ruby
%i(ensure purged).include?(@resource[:ensure])
```

Reply:
@ciprianbadescu `hash` contains the resource queried from the system, with `dpkg`. I am interested in the requested `@resource` state.